app/controllers/media_controller.rb

```diff
@@ -76,6 +74,22 @@ def requested_from_draft_assets_host?
     request.host == AssetManager.govuk.draft_assets_host
   end
 
+  def draft_asset_manager_access?
+    token = params.fetch(:token, cookies[:auth_bypass_token])
+    payload = decode_auth_bypass_token(token)
```

Comment:
It might be worth having this set as part of a before action or in a method of it's own so it doesn't need to be replicated. You could have something like:
```
before_action :set_token_payload

def set_token_payload
  token = params.fetch(:token, cookies[:auth_bypass_token])
  @token_payload = if token
                     secret = Rails.application.secrets.jwt_auth_secret
                     JWT.decode(token, secret, true, algorithm: "HS256").first
                   end
rescue JWT::DecodeError
  @token_payload = nil
end
```